db/blip_handler.go

```diff
@@ -405,8 +411,8 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
 		// message, or a continuous replication catches up *for the first time*.
 		// Note that this doesn't mean that rev messages associated with previous changes
 		// messages have been fully processed
-		if bh.emptyChangesMessageCallback != nil {
-			bh.emptyChangesMessageCallback()
+		if bh.onReplicationComplete != nil {
```

Comment:
The problem with using onReplicationComplete here is this isn't necessarily a complete replication - for a continuous replication, it's just the first time it's caught up.  We're only setting onReplicationComplete for pull in the one-shot case today, but I feel like making this change introduces the potential for future regressions where we set onReplicationComplete for continuous pull replications, and then have it fire unexpectedly.

Reply:
So either revert to the previous approach (separate callbacks), or If you want to use onReplicationComplete here, do a check to see if the replication is one-shot before invoking the callback.